lib/questionlib.php

```diff
@@ -1764,17 +1764,23 @@ function question_edit_url($context) {
  * this method will help to autoload those nodes in the question bank navigation.
  *
  * @param navigation_node $navigationnode The navigation node to add the question branch to
- * @param object $context
+ * @param context $context
  * @param string $baseurl the url of the base where the api is implemented from
+ * @param bool $default If use the default navigation or needs id as parameter.
  * @return navigation_node Returns the question branch that was added
  */
-function question_extend_settings_navigation(navigation_node $navigationnode, $context, $baseurl = '/question/edit.php') {
+function question_extend_settings_navigation(navigation_node $navigationnode, $context, $baseurl = '/question/edit.php', $default = true) {
     global $PAGE;
 
     if ($context->contextlevel == CONTEXT_COURSE) {
         $params = ['courseid' => $context->instanceid];
     } else if ($context->contextlevel == CONTEXT_MODULE) {
         $params = ['cmid' => $context->instanceid];
+        if ($default) {
+            $paramqbank = ['cmid' => $context->instanceid];
```

Comment:
if default is true, do we need this logic? its same as $param, so if its true, cant we use $param?

Reply:
As we need both, $paramqbank and $params (this last one for categories, export and import) so we can add the logic in each use of id or cmid depending of default or we can have it there and only do the validation once.